### ISSUE TITLE
List single list item does not work

### ISSUE THREAD

**Issue by Matt-Butler** [cerner/terra-core#425]
# Lists do not work with a single list item.

```
import React from 'react';
import SingleSelectList from 'terra-list/lib/SingleSelectList';

const myURL = 'http://www.google.com';
const myContent = <a href={myURL}><span>MyText</span></a>;

const list = () => (
  <SingleSelectList isDivided={true} hasChevrons={false}>
     <SingleSelectList.Item content={myContent} key="myKey" isSelectable />
  </SingleSelectList>
);



export default list;

```
![screen shot 2017-06-01 at 12 13 47 pm](https://cloud.githubusercontent.com/assets/8146159/26691719/cddaa258-46c3-11e7-8250-4e697c64ee53.png)

## Issue Type

- [ ] New Feature
- [ ] Enhancement
- [x] Bug
- [ ] Other

## Expected Behavior
List should work with a single list item

## Current Behavior

List does not work with a single list item

## Steps to Reproduce

1. Copy the code above 
or 
1. Checkout this branch: https://github.com/cerner/terra-core/tree/list-single-item. It is the last list example.

**Comment by Matt-Butler** [comment]
The issue is with the cloneChildItems method, it expects and array of listItem objects. It can actually be an empty array or a single listItem object.
https://github.com/cerner/terra-core/blob/master/packages/terra-list/src/SingleSelectList.jsx#L73

Update thecloneChildItems method to this:
```
  cloneChildItems(items) {
    if (items === []) {
      return items;
    }

    if (items.constructor != Array) {
      const wrappedOnClick = this.wrappedOnClickForItem(items, 0);
      const wrappedOnKeyDown = this.wrappedOnKeyDownForItem(items, 0);
      const newProps = this.newPropsForItem(items, 0, wrappedOnClick, wrappedOnKeyDown);

      return React.cloneElement(items, newProps);
    }

    return items.map((item, index) => {
      const wrappedOnClick = this.wrappedOnClickForItem(item, index);
      const wrappedOnKeyDown = this.wrappedOnKeyDownForItem(item, index);
      const newProps = this.newPropsForItem(item, index, wrappedOnClick, wrappedOnKeyDown);

      return React.cloneElement(item, newProps);
    });
  }
```

Do the same for list and multiselect.

**Comment by bjankord** [comment]
Released in terra-list@0.4.0
